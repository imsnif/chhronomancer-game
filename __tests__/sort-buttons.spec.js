import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import SortButtons from '../components/sort-buttons'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

test('SortButtons (sortBy player) => renders correctly', () => {
  const tree = renderer.create(
    <SortButtons timelineStore={{sortBy: 'player'}}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SortButtons (sortBy type) => renders correctly', () => {
  const tree = renderer.create(
    <SortButtons timelineStore={{sortBy: 'type'}}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SortButtons (sortBy time) => renders correctly', () => {
  const tree = renderer.create(
    <SortButtons timelineStore={{sortBy: 'time'}}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SortButtons (sortBy player) => press player', () => {
  const changeSort = jest.fn()
  const wrapper = shallow(
    <SortButtons timelineStore={{sortBy: 'player', changeSort}}/>
  )
  const button = wrapper.find('TouchableHighlight').at(0)
  button.simulate('press')
  expect(changeSort).toBeCalledWith('player')
})

test('SortButtons (sortBy type) => press type', () => {
  const changeSort = jest.fn()
  const wrapper = shallow(
    <SortButtons timelineStore={{sortBy: 'player', changeSort}}/>
  )
  const button = wrapper.find('TouchableHighlight').at(1)
  button.simulate('press')
  expect(changeSort).toBeCalledWith('type')
})

test('SortButtons (sortBy time) => press time', () => {
  const changeSort = jest.fn()
  const wrapper = shallow(
    <SortButtons timelineStore={{sortBy: 'player', changeSort}}/>
  )
  const button = wrapper.find('TouchableHighlight').at(2)
  button.simulate('press')
  expect(changeSort).toBeCalledWith('time')
})
