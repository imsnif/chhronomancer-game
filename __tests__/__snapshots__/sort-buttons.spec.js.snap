// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButtons (sortBy player) => renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "#329932",
          "flex": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#0e0e0e",
          "fontFamily": "telegrama_raw",
          "fontSize": 20,
          "paddingTop": 13,
          "textAlign": "center",
        }
      }
    >
      Player
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "#141313",
          "borderBottomWidth": 1,
          "borderColor": "#329932",
          "borderRadius": 0,
          "flex": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#329932",
          "fontFamily": "telegrama_raw",
          "fontSize": 20,
          "paddingTop": 13,
          "textAlign": "center",
        }
      }
    >
      Type
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "#141313",
          "borderColor": "#329932",
          "borderRadius": 0,
          "flex": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#329932",
          "fontFamily": "telegrama_raw",
          "fontSize": 20,
          "paddingTop": 13,
          "textAlign": "center",
        }
      }
    >
      Time
    </Text>
  </View>
</View>
`;

exports[`SortButtons (sortBy time) => renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "#329932",
          "flex": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#0e0e0e",
          "fontFamily": "telegrama_raw",
          "fontSize": 20,
          "paddingTop": 13,
          "textAlign": "center",
        }
      }
    >
      Player
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "#141313",
          "borderBottomWidth": 1,
          "borderColor": "#329932",
          "borderRadius": 0,
          "flex": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#329932",
          "fontFamily": "telegrama_raw",
          "fontSize": 20,
          "paddingTop": 13,
          "textAlign": "center",
        }
      }
    >
      Type
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "#141313",
          "borderColor": "#329932",
          "borderRadius": 0,
          "flex": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#329932",
          "fontFamily": "telegrama_raw",
          "fontSize": 20,
          "paddingTop": 13,
          "textAlign": "center",
        }
      }
    >
      Time
    </Text>
  </View>
</View>
`;

exports[`SortButtons (sortBy type) => renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "#329932",
          "flex": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#0e0e0e",
          "fontFamily": "telegrama_raw",
          "fontSize": 20,
          "paddingTop": 13,
          "textAlign": "center",
        }
      }
    >
      Player
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "#141313",
          "borderBottomWidth": 1,
          "borderColor": "#329932",
          "borderRadius": 0,
          "flex": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#329932",
          "fontFamily": "telegrama_raw",
          "fontSize": 20,
          "paddingTop": 13,
          "textAlign": "center",
        }
      }
    >
      Type
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "#141313",
          "borderColor": "#329932",
          "borderRadius": 0,
          "flex": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#329932",
          "fontFamily": "telegrama_raw",
          "fontSize": 20,
          "paddingTop": 13,
          "textAlign": "center",
        }
      }
    >
      Time
    </Text>
  </View>
</View>
`;
